dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-fee-record-correction-review.controller/helpers.ts

```diff
@@ -0,0 +1,95 @@
+  }
+}
+
+/**
+ * Gets the formatted value from form data for a specific correction reason.
+ * @param formData - The transient form data containing corrected values
+ * @param reason - The reason for the correction
+ * @returns The formatted value corresponding to the correction reason
+ * @throws Error if the correction reason is not supported
+ */
+export const getFormattedFormDataValueForCorrectionReason = (formData: RecordCorrectionTransientFormData, reason: RecordCorrectionReason) => {
+  switch (reason) {
+    case RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT:
+      return formData.facilityId;
+    case RECORD_CORRECTION_REASON.REPORTED_CURRENCY_INCORRECT:
+      return formData.reportedCurrency;
+    case RECORD_CORRECTION_REASON.REPORTED_FEE_INCORRECT:
+      validateRequiredFormDataField(formData.reportedFee, reason);
+
+      return getFormattedMonetaryValue(formData.reportedFee);
+    case RECORD_CORRECTION_REASON.UTILISATION_INCORRECT:
+      validateRequiredFormDataField(formData.utilisation, reason);
+
+      return getFormattedMonetaryValue(formData.utilisation);
+    case RECORD_CORRECTION_REASON.OTHER:
+      return '-';
+    default:
+      throw new Error(`Unsupported record correction reason: ${reason}`);
+  }
+};
+
+/**
+ * Maps through an array of correction reasons and returns an array of
+ * formatted values from the transient form data.
+ * @param formData - The transient form data containing corrected values
+ * @param reasons - Array of reasons for the correction
+ * @returns Array of formatted values from the form data, corresponding to each correction reason
+ */
+export const mapFormDataToFormattedValues = (formData: RecordCorrectionTransientFormData, reasons: RecordCorrectionReason[]) => {
+  return reasons.map((reason) => getFormattedFormDataValueForCorrectionReason(formData, reason));
+};
+
+/**
+ * Maps transient form data and fee record correction entity into review information.
+ * @param formData - The transient form data containing the corrected values
+ * @param feeRecordCorrection - The fee record correction entity containing original record and correction details
+ * @returns Formatted review information
+ */
+export const mapTransientCorrectionDataToReviewInformation = (
+  formData: RecordCorrectionTransientFormData,
+  feeRecordCorrection: FeeRecordCorrectionEntity,
+): FeeRecordCorrectionReviewInformation => {
+  const { id: correctionId, feeRecord, reasons, additionalInfo: errorSummary } = feeRecordCorrection;
+
+  const mappedFeeRecord = {
+    exporter: feeRecord.exporter,
+    reportedFees: {
+      currency: feeRecord.feesPaidToUkefForThePeriodCurrency,
+      amount: feeRecord.feesPaidToUkefForThePeriod,
+    },
+  };
+
+  const formattedOldValues = mapCorrectionReasonsToFormattedOldValues(feeRecord, reasons);
+  const formattedNewValues = mapFormDataToFormattedValues(formData, reasons);
```

Comment:
I think it is a little bit confusing that these have the same name as the fields in the returned type but different types. Could we perhaps join immediately, or rename to have 'array' at the end of the name?